src/include/daos_fs_sys.h

```diff
@@ -0,0 +1,454 @@
+ */
+int
+dfs_sys_chmod(dfs_sys_t *dfs_sys, const char *path, mode_t mode);
+
+/**
+ * set stat attributes for a file and fetch new values.
+ * By default, if the object is a symlink the link itself is modified.
+ * See dfs_sys_stat() for which entries are filled.
+ *
+ * \param[in]	dfs_sys	Pointer to the mounted file system.
+ * \param[in]	path	Link path of object.
+ * \param[in,out]
+ *		stbuf	[in]: Stat struct with the members set.
+ *			[out]: Stat struct with all valid members filled.
+ * \param[in]	flags	Bitmask of flags to set.
+ * \param[in]	sflags	(O_NOFOLLOW)
+ *
+ * \return		0 on Success. errno code on Failure.
+ */
+int
+dfs_sys_setattr(dfs_sys_t *dfs_sys, const char *path, struct stat *stbuf,
+		int flags, int sflags);
+
+/**
+ * Set atime and mtime of a path.
+ *
+ * \param[in]	dfs_sys Pointer to the mounted file system.
+ * \param[in]	path	Link path of object.
+ * \param[in]	times	[0]: atime to set
+ *			[1]: mtime to set
+ * \param[in]	flags	(O_NOFOLLOW)
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_utimens(dfs_sys_t *dfs_sys, const char *path,
+		const struct timespec times[2], int flags);
+
+/**
+ * stat attributes of an entry. By default, if object is a symlink,
+ * the link itself is interrogated. The following elements of the
+ * stat struct are populated (the rest are set to 0):
+ * mode_t	st_mode;
+ * uid_t	st_uid;
+ * gid_t	st_gid;
+ * off_t	st_size;
+ * blkcnt_t	st_blocks;
+ * struct timespec st_atim;
+ * struct timespec st_mtim;
+ * struct timespec st_ctim;
+ *
+ * \param[in]	dfs_sys Pointer to the mounted file system.
+ * \param[in]	path	Link path of object.
+ * \param[in]	stbuf	Stat struct with the members above filled.
+ * \param[in]	flags	Stat flags (O_NOFOLLOW).
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_stat(dfs_sys_t *dfs_sys, const char *path, struct stat *buf,
+	     int flags);
+/**
+ * Create a file or directory.
+ *
+ * \param[in]	dfs_sys Pointer to the mounted file system.
+ * \param[in]	path	Link path of new object.
+ * \param[in]	mode	mode_t (permissions + type).
+ * \param[in]	cid	DAOS object class id (pass 0 for default MAX_RW).
+ *			Valid on create only; ignored otherwise.
+ * \param[in]	chunk_size
+ *			Chunk size of the array object to be created.
+ *			(pass 0 for default 1 MiB chunk size).
+ *			Valid on file create only; ignored otherwise.
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_mknod(dfs_sys_t *dfs_sys, const char *path, mode_t mode,
+	      daos_oclass_id_t cid, daos_size_t chunk_size);
+
+/**
+ * list extended attributes of a path and place them all in a buffer
+ * NULL terminated one after the other. By default, if path is a
+ * symlink, the link itself is interrogated.
+ *
+ * \param[in]	dfs_sys Pointer to the mounted file system.
+ * \param[in]	path	Link path of object.
+ * \param[in,out]
+ *		list	[in]: Allocated buffer for all xattr names.
+ *			[out]: Names placed after each other (null terminated).
```

Comment:
I would recommend running these through doxygen to see how they render.  I suspect not very well.

Reply:
On that note, you may need to add this file to the doxygen config file.

Reply:
FWIW, I did copy most of this from daos_fs.h